src/js/models/menubar.js

```diff
@@ -260,19 +260,63 @@ define(function (require, exports, module) {
     };
 
     /**
+     * Replace keyChar and keyCode properties of menu shortcuts with localized
+     * strings.
+     *
+     * @private
+     * @param {Array.<object>} entries
+     * @param {Object.<string, string>} shortcuts
+     */
+    var _resolveShortcuts = function (entries, shortcuts) {
+        entries.forEach(function (entry) {
+            var id = entry.id;
+
+            if (entry.hasOwnProperty("shortcut")) {
+                var shortcut = entry.shortcut;
+                if (!shortcut.keyCode && !shortcut.keyChar) {
+                    throw new Error("Menu entry " + id + " has a shortcut without a key.");
+                }
+
+                if (!shortcuts) {
```

Comment:
This a little convoluted but intentional. The point is that it's OK if there is no localization dictionary for a particular submenu ID _only if_ there are no entries that have keyboard shortcuts. So, it's an error as soon as you have a shortcut key that needs to be localized. Yes, I could do a first pass over the array and check for the existence of shortcuts first, but this seemed just as well to me...